pkg/util/template/resource.go

```diff
@@ -67,41 +94,73 @@ type PlainText struct {
 	ComponentDependencies []*PlainText
 }
 
+type MessagePlainText struct {
+	// Name is the name of template
+	Name string
+}
+
 var _ resource.Descriptor = &PlainText{}
+var _ resource.Descriptor = &MessagePlainText{}
 
-func (t *PlainText) templateResource() {}
+func (t *PlainText) templateResource()        {}
+func (t *MessagePlainText) templateResource() {}
 
 func (t *PlainText) MatchResource(path string) (*resource.Match, bool) {
+	if strings.HasPrefix(t.Name, "messages/") {
+		return nil, false
+	}
+	return matchTemplatePath(path, t.Name)
+}
+
+func (t *MessagePlainText) MatchResource(path string) (*resource.Match, bool) {
+	if !strings.HasPrefix(t.Name, "messages/") {
+		return nil, false
+	}
 	return matchTemplatePath(path, t.Name)
 }
 
 func (t *PlainText) FindResources(fs resource.Fs) ([]resource.Location, error) {
+	// Exclude App Fs
+	if fs.GetFsLevel() == resource.FsLevelApp {
+		return []resource.Location{}, nil
+	}
```

Comment:
PlainText originally does not have FindResourcesFsFilter, so we have to preserve the original behavior. Allow it to read from any Fs. Please remove this condition.

Reply:
removed